### PR TITLE
Refund on unused native tokens

### PR DESCRIPTION
**Pull request by ocnc**

### CODE REVIEW

File: contracts/periphery/BeraCrocMultiSwap.sol [hunk 1]
```diff
@@ -160,11 +159,19 @@ contract BeraCrocMultiSwap {
...
     }
 
-    function retire() external {
-        require(msg.sender == _deployer, "Only deployer can retire");
-        // drain the honey and stgusdc
-        IERC20Minimal(0x7EeCA4205fF31f947EdBd49195a7A88E6A91161B).transfer(_deployer, IERC20Minimal(0x7EeCA4205fF31f947EdBd49195a7A88E6A91161B).balanceOf(address(this)));
-        IERC20Minimal(0x6581e59A1C8dA66eD0D313a0d4029DcE2F746Cc5).transfer(_deployer, IERC20Minimal(0x6581e59A1C8dA66eD0D313a0d4029DcE2F746Cc5).balanceOf(address(this)));
-        selfdestruct(payable(_deployer));
+    /* @notice Refund any remaining Ether in the contract to the sender. */
+    function _refund() internal {
```
Comment (calbera) [inline contracts/periphery/BeraCrocMultiSwap.sol]:
```suggestion
    function _refundValue() internal {
```
to be more explicit

Comment (calbera) [reply contracts/periphery/BeraCrocMultiSwap.sol]: or `refundEther` or something like that

File: contracts/periphery/BeraCrocMultiSwap.sol [hunk 2]
```diff
@@ -160,11 +159,19 @@ contract BeraCrocMultiSwap {
...
+        if (address(this).balance > 0) {
+            payable(msg.sender).transfer(address(this).balance);
+        }
+    }
+
+    /* @notice // Prevent direct transfers to the contract by reverting any Ether sent directly. */
+    receive() external payable {
```
Comment (sj448) [inline contracts/periphery/BeraCrocMultiSwap.sol]: can remove receive and fallback functions to achieve the same behaviour

**Review by calbera — APPROVED** [review]

File: contracts/periphery/BeraCrocMultiSwap.sol [hunk 3]
```diff
@@ -160,11 +159,19 @@ contract BeraCrocMultiSwap {
...
-    function retire() external {
-        require(msg.sender == _deployer, "Only deployer can retire");
-        // drain the honey and stgusdc
-        IERC20Minimal(0x7EeCA4205fF31f947EdBd49195a7A88E6A91161B).transfer(_deployer, IERC20Minimal(0x7EeCA4205fF31f947EdBd49195a7A88E6A91161B).balanceOf(address(this)));
-        IERC20Minimal(0x6581e59A1C8dA66eD0D313a0d4029DcE2F746Cc5).transfer(_deployer, IERC20Minimal(0x6581e59A1C8dA66eD0D313a0d4029DcE2F746Cc5).balanceOf(address(this)));
-        selfdestruct(payable(_deployer));
+    /* @notice Refund any remaining Ether in the contract to the sender. */
+    function _refund() internal {
+        if (address(this).balance > 0) {
+            payable(msg.sender).transfer(address(this).balance);
```
Comment (sj448) [inline contracts/periphery/BeraCrocMultiSwap.sol:165]: vulnerable to reentrancy attack,  msg.sender being a smart contract can call it back/DOS too 
maybe, did we check those cases?

Comment (calbera) [reply contracts/periphery/BeraCrocMultiSwap.sol:165]: uniswap does [this](https://github.com/Uniswap/v3-periphery/blob/697c2474757ea89fec12a4e6db16a574fe259610/contracts/libraries/TransferHelper.sol#L56) for its [safeTransfer](https://github.com/Uniswap/v3-periphery/blob/697c2474757ea89fec12a4e6db16a574fe259610/contracts/base/PeripheryPayments.sol#L45)

Comment (ocnc) [reply contracts/periphery/BeraCrocMultiSwap.sol:165]: ~~I'll switch it to use the safeTransfer, but i also don't think theres a reentrancy attack as this full yoinks whatever remaining eth exists anyways. I don't see what else there would be to take~~

re: barny's next comment

Comment (sj448) [reply contracts/periphery/BeraCrocMultiSwap.sol:165]: yea transfer should be good it protects from re-entrancy due to fixed gas.

Comment (sj448) [reply contracts/periphery/BeraCrocMultiSwap.sol:165]: @calbera uniswap refundETH  uses .call to send ether
tranfer is not recommended method post Istanbul hard-fork, we should change it to use .call like uniswap and should be good then.
i agree with @ocnc that there shouldn't be any case of re-entrancy as no funds will be left.
https://consensys.io/diligence/blog/2019/09/stop-using-soliditys-transfer-now/

Comment (ocnc) [reply contracts/periphery/BeraCrocMultiSwap.sol:165]: ill make that change 👍

**Pull request merged**